resources/lib/player/players.py

```diff
@@ -145,17 +146,27 @@ def _set_external_ids(self, tmdb_type, tmdb_id, season, episode, details, requir
         return set_detailed_item(tmdb_type, tmdb_id, season, episode, details=details) or {}
 
     def _set_provider_priority(self):
-        try:  # Check if players are listed in providers and bump priority if found
-            providers = self.details.infoproperties['providers']
-            providers = providers.split(' / ')
-            for k, v in self.players.items():
-                if 'provider' not in v or v['provider'] not in providers:
-                    v['priority'] = v.get('priority', PLAYERS_PRIORITY) + 100  # Increase priority baseline by 100 to prevent other players displaying above providers
-                    continue
-                v['priority'] = providers.index(v['provider']) + 1  # Add 1 because sorted() puts 0 index last
-                v['is_provider'] = True
-        except (KeyError, AttributeError):
-            pass
+        try:
+            providers = self.details.infoproperties.get('providers', '')
+        except AttributeError:
+            providers = ''
+        providers = tuple(providers.split(' / '))
+
+        def _set_priority(item, _providers=providers):
+            file, player = item
+            player_provider = player.get('provider')
+            if player_provider in _providers:
+                priority = _providers.index(player_provider) + 1  # Add 1 because sorted() puts 0 index last
+                player['is_provider'] = True
+            else:
+                if player.get('is_provider', True):
+                    priority = player.get('priority', PLAYERS_PRIORITY) + 100  # Increase priority baseline by 100 to prevent other players displaying above providers
+                player['is_provider'] = False    
+            player['priority'] = priority
+            return priority, u_normalize('NFKD', player.get('name', '\uFFFF')).lower()
```

Comment:
The intent was to have players with the same priority sorted in an easily understandable manner (alphabetically) in the player select dialog.

The original filename sort may be off because of the "autogen" prefix, or simply because the file naming convention is not specified anywhere. From what I have seen the filename for players may the provider name, others may be the plugin name e.g. "plugin.xxx.yyy", others may have other various prefixes e.g. "direct".

I didn't consider that the player name may have BBCode in it, which would throw things off in much the same way as the differing file naming conventions would.

In terms of the normalisation, that was just to deal with a hypothetical scenario where the player names included non-ASCII characters, so that a player name starting with `Ä`, or any other Unicode character, was not sorted by default after all the ASCII names.

Perhaps an alternative that addresses all four issues - different file naming conventions, BBCode in player names, non-ASCII characters, and potentially slow normalisation - is to just use the plugin name as the second sort key, as that is shown as label2 in the dialog entries?

Or I can just remove the second sort key.

Reply:
9f8e7ec removes the normalisation and uses plugin name as a secondary sort key

Reply:
Probably not worth the trouble trying to come up with foolproof solution. I think on the balance, plugin name is a good enough fallback.

Easy enough for users to adjust priority according to their own preferences via the customisation dialog.